### ISSUE TITLE
Idle Time feature request

### ISSUE THREAD

**Issue by Lancelight** [vertcoin-project/one-click-miner-vnext#120]
One critical feature that is missing in the OCM for normal people (like myself) to really make it useful is the ability to have it turn on/off by itself based on an "Idle time".  In windows task scheduler, the "idle time" event does NOT work (it never has and it's been a problem for years there are posts about it all over the place),  neither in Win10 nor in Win7.  So a custom idle time function would need to be created as relying upon the idle time functions in windows may not work as expected.

**Comment by gertjaap** [comment]
Is this a duplicate of #57  ?

**Comment by Lancelight** [comment]
It's not a duplicate but it's a similar request.  Idle scheduling mining IMO is more efficient since hours at a computer may vary.  But scheduled mining would be great also.  Similar,  but different features.

**Comment by gertjaap** [comment]
So your suggestion means we have to think out and code up a protocol to determine if the user is idle all from scratch? That seems far out of reach for this project to be honest, with very limited upwards benefit.

**Comment by Lancelight** [comment]
Yes it very could be for the windows platform.  The idle checking functions for Linux are likely to work just fine out of the box.  But as far as Winblows goes..... It could be hit or miss.  They might work, but then again,  they might not.  As for limited Benefit,  consider this.  I spend maybe 15-30 mins of my day on discord and probably half that time is spent looking at the "general" channel for VTC.  Over the last 5 days,  I have seen no less than 3 people ask for this very feature and thats with the limited time I spend there.  So I wouldnt say there is limited benefit if people are looking for something like this.

**Comment by Lancelight** [comment]
Here are the two functions that would need to be used for both platforms:
https://linux.die.net/man/3/xscreensaverqueryinfo
https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getlastinputinfo?redirectedfrom=MSDN

**Comment by phearsom** [comment]
Large problem with letting a process to mine when a user is "idle" is whether or not you take into consideration whether or not the system is idle. Another thing suspicious about this type of work is letting a service start up all by itself. Literally a definition of a trojan. It's better intention to leave it as simple as 1 click mining. No other requirement for easy mode.

**Comment by Lancelight** [comment]
Thats probably the worst reason to leave out important features.  If items
were always created using that method, we would still be using clubs and
spears to harvest food lol.  Anything can be used improperly,  including
the coin itself.  There are actually more legit reasons for this type of
feature than some conspiracy theory that it might be used as a trojan.
This request isnt asking for stealth features.  Its asking for idle
features which is completely different.  A trojan is no good if you can SEE
its running.
